modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/ResolverSetUtils.java

```diff
@@ -195,10 +230,14 @@ private static Map<String, ValueResolver> getParameterGroupParametersValueResolv
   private static ValueResolver getParameterValueResolver(String parameterName, MetadataType type, Object value,
                                                          Set<ModelProperty> modelProperties, ReflectionCache reflectionCache,
                                                          MuleContext muleContext, ValueResolverFactory valueResolverFactory,
-                                                         boolean acceptsReferences)
+                                                         boolean acceptsReferences, ApplicationTypeLoader applicationTypeLoader)
       throws MuleException {
+    if (value instanceof EnrichedValue) {
```

Comment:
I don't understand why you need the EnrichedValue at all, but this is a code smell and something that can very easily turn into a bug. Sooner or later the code is going to change, one of this `instanceof` is going to be missing and this will return an `EnrichedValue` instead of the actual one.